pyrsa/data/dataset.py

```diff
@@ -0,0 +1,165 @@
+        elif (measurements.ndim == 3):
+            self.measurements = measurements
+            self.n_set, self.n_obs, self.n_channel = self.measurements.shape
+        self.descriptors = descriptors
+        self.obs_descriptors = obs_descriptors
+        self.channel_descriptors = channel_descriptors
+
+    def split_obs(self, by):
+        """ Returns a list Datasets splited by obs
+        Args:
+            by(String): the descriptor by which the splitting is made
+
+        Returns:
+            list of Datasets, splitted by the selected obs_descriptor
+        """
+        raise NotImplementedError(
+            "split_obs function not implemented in used Dataset class!"
+        )
+
+    def split_channel(self, by):
+        """ Returns a list Datasets splited by channels
+        Args:
+            by(String): the descriptor by which the splitting is made
+
+        Returns:
+            list of Datasets,  splitted by the selected channel_descriptor
+        """
+        raise NotImplementedError(
+            "split_channel function not implemented in used Dataset class!"
+        )
+
+    def subset_obs(self, by, value):
+        """ Returns a subsetted Dataset defined by certain obs value
+        Args:
+            by(String): the descriptor by which the subset selection is made
+                        from obs dimension
+            value:      the value by which the subset selection is made
+                        from obs dimension
+
+        Returns:
+            Dataset, with subset defined by the selected obs_descriptor
+        """
+        raise NotImplementedError(
+            "subset_obs function not implemented in used Dataset class!"
+        )
+
+    def subset_channel(self, by, value):
+        """ Returns a subsetted Dataset defined by certain channel value
+        Args:
+            by(String): the descriptor by which the subset selection is made
+                        from channel dimension
+            value:      the value by which the subset selection is made
+                        from channel dimension
+
+        Returns:
+            Dataset, with subset defined by the selected channel_descriptor
+        """
+        raise NotImplementedError(
+            "subset_channel function not implemented in used Dataset class!"
+        )
+
+
+class Dataset(DatasetBase):
+    """
+    Dataset class is a standard version of DatasetBase.
+    It contains one data set - or multiple data sets with the same structure
+    """
+    def split_obs(self, by):
+        """ Returns a list Datasets splited by obs
+        Args:
+            by(String): the descriptor by which the splitting is made
+
+        Returns:
+            list of Datasets, splitted by the selected obs_descriptor
+        """
+        unique_values = set(self.obs_descriptors[by])
+        dataset_list = []
+        for v in unique_values:
+            dataset_list.append(
+                self.measurements[:, self.obs_descriptors[by] == v, :])
```

Comment:
For future reference, another way to create shorter lines, is to break up the statement by creating a separate (local scope) indexing variable. For instance:

```
selection = self.obs_descriptors[by] == v, :]
dataset_list.append(self.measurements[:, selection])

```

Reply:
Good idea! Will change similar structure and push next week.

Reply:
fixed and pushed.